File: src/test/java/com/robohorse/robopojogenerator/generator/ClassCreatorTest.java
package com.robohorse.robopojogenerator.generator;

import com.robohorse.robopojogenerator.generator.common.ClassCreator;
import com.robohorse.robopojogenerator.models.ClassItemModel;
import com.robohorse.robopojogenerator.models.GenerationModel;
import com.robohorse.robopojogenerator.models.ProjectModel;
import com.robohorse.robopojogenerator.delegates.FileWriterDelegate;
import org.junit.Before;
import org.junit.Test;
import org.mockito.InjectMocks;
import org.mockito.Mock;
import org.mockito.MockitoAnnotations;

import java.util.HashSet;
import java.util.Set;

import static org.mockito.Mockito.verify;
import static org.mockito.Mockito.when;

/**
 * Created by vadim on 22.10.16.
 */
public class ClassCreatorTest {
    @InjectMocks
    ClassCreator classCreator;
    @Mock
    RoboPOJOGenerator roboPOJOGenerator;
    @Mock
    FileWriterDelegate fileWriterDelegate;

    @Before
    public void setUp() {
        MockitoAnnotations.initMocks(this);
    }

    @Test
    public void generateFiles() throws Exception {
        final ClassItemModel classItemModel = new ClassItemModel("");
        final Set<ClassItemModel> classItemModelSet = new HashSet<ClassItemModel>();
        classItemModelSet.add(classItemModel);
        final GenerationModel generationModel = new GenerationModel
                .Builder()
                .build();
        final ProjectModel projectModel = new ProjectModel
                .Builder()
                .build();
        when(roboPOJOGenerator.generate(generationModel))
                .thenReturn(classItemModelSet);
        classCreator.generateFiles(generationModel, projectModel);
        verify(fileWriterDelegate)
                .writeFile(classItemModel, generationModel, projectModel);
    }
}

File: src/test/kotlin/com/robohorse/robopojogenerator/generator/ClassCreatorTest.kt
package com.robohorse.robopojogenerator.generator

import com.robohorse.robopojogenerator.generator.consts.common.ClassCreator
import com.robohorse.robopojogenerator.generator.consts.common.ClassItem
import com.robohorse.robopojogenerator.generator.consts.common.FileWriteFactory
import com.robohorse.robopojogenerator.models.GenerationModel
import com.robohorse.robopojogenerator.models.ProjectModel
import io.mockk.MockKAnnotations
import io.mockk.every
import io.mockk.impl.annotations.InjectMockKs
import io.mockk.impl.annotations.RelaxedMockK
import io.mockk.verify
import org.junit.Test
import kotlin.test.BeforeTest

class ClassCreatorTest {

    @RelaxedMockK
    lateinit var roboPOJOGenerator: RoboPOJOGenerator

    @RelaxedMockK
    lateinit var fileWriteFactory: FileWriteFactory

    @RelaxedMockK
    lateinit var generationModel: GenerationModel

    @RelaxedMockK
    lateinit var projectModel: ProjectModel

    @InjectMockKs
    lateinit var classCreator: ClassCreator

    @BeforeTest
    fun setUp() = MockKAnnotations.init(this, relaxUnitFun = true)

    @Test
    fun generateFiles() {
        val classItem = ClassItem("")
        val classItemSet: MutableSet<ClassItem> = HashSet()
        classItemSet.add(classItem)
        every { roboPOJOGenerator.generate(generationModel) }.returns(classItemSet)
        classCreator.generateFiles(generationModel, projectModel)
        verify { fileWriteFactory.createFileWriter(generationModel) }
    }
}
